testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/secondlevelcache/SFSB4LC.java

```diff
@@ -0,0 +1,112 @@
+package org.jboss.as.test.integration.jpa.secondlevelcache;
+
+import org.hibernate.Session;
+import org.hibernate.stat.QueryStatistics;
+import org.hibernate.stat.Statistics;
+
+import javax.ejb.Stateless;
+import javax.ejb.TransactionAttribute;
+import javax.ejb.TransactionAttributeType;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.PersistenceUnit;
+import javax.persistence.Query;
+
+import static org.junit.Assert.assertEquals;
+
+@Stateless
+public class SFSB4LC {
+
+    @PersistenceUnit(unitName = "mypc")
+    EntityManagerFactory emf;
+
+    /**
+     * Create employee in provided EntityManager
+     */
+    public void createEmployee(String name, String address, int id) {
+        EntityManager em = emf.createEntityManager();
+        Employee emp = new Employee();
+        emp.setId(id);
+        emp.setAddress(address);
+        emp.setName(name);
+        try {
+            em.persist(emp);
+            em.flush();
+        } catch (Exception e) {
+            throw new RuntimeException("transactional failure while persisting employee entity", e);
+        }
+    }
+
+    /**
+     * Performs 2 query calls, first call put entity in the cache and second should hit the cache
+     *
+     * @param id Employee's id in the query
+     */
+    public String entityCacheCheck(int id) {
+
+        // the nextTimestamp from infinispan is "return System.currentTimeMillis()"
+        try {
+            Thread.sleep(1000);
```

Comment:
Why this sleep? It is not clear to me from comment why it is here.

Reply:
@simkam I don't know: I copied this piece of code from the existing class SFSB2LC; maybe it is for timestamps

Reply:
The SFSB2LC sleep(1000) code was added in 2012, so its been a while but from what I recall it compensated for Infinispan internals, that avoided a test failure.  @zroubalik do you recall?

Reply:
It's been a while, I'm not excatly sure, but from what I recall @scottmarlow is right. It should do a compensation for Infinispan internals. Sometimes the test was too quick and Infinispan wasn't in sync and caused some test failures. I'd recommend to try this test **several** times without the sleep to check if there aren't any intermittent failures.